pytorch_lightning/plugins/training_type/ddp_spawn.py

```diff
@@ -415,3 +419,11 @@ def teardown(self) -> None:
             self.lightning_module.cpu()
             # clean up memory
             torch.cuda.empty_cache()
+
+    @staticmethod
+    def _clean_logger(trainer: "pl.Trainer") -> None:
+        loggers = trainer._logger_iterable if isinstance(trainer.logger, LoggerCollection) else [trainer.logger]
+        for logger in loggers:
+            if isinstance(logger, TensorBoardLogger) and logger._experiment is not None:
+                rank_zero_warn("When using `ddp_spawn`, the Tensorboard experiment should be `None`.")
+                logger._experiment = None
```

Comment:
This doesn't guarantee that garbage collection will cleanup the writer instance (where I expect the spawn queue gets closed)

Should we del instead?

```suggestion
                del logger._experiment
                logger._experiment = None
```

or

```suggestion
                logger._experiment = None
                gc.collect()
```

Reply:
Done !